tutorial/guias/Paso_a_paso_DIA_6.md

```diff
@@ -0,0 +1,375 @@
+            else:
+                self.snake.trim()
+
+            Renderer.render_game_state(
+                self.snake.body,
+                self.fruit.position,
+                self.game_window,
+                Color.GREEN.value,
+                Color.WHITE.value,
+                Color.BLACK.value,
+            )
+
+            if self.game_over():
+                Renderer.render_game_over(
+                    self.score,
+                    self.game_window,
+                    self.window_x,
+                    self.window_y,
+                    Color.RED.value,
+                    "times new roman",
+                    50,
+                )
+                time.sleep(2)
+                pygame.quit()
+                sys.exit()
+
+            Renderer.render_score(
+                self.score, self.game_window, Color.WHITE.value, "times new roman", 20
+            )
+            pygame.display.update()
+            self.fps.tick(self.snake_speed)
+```
+
+La clase `Game` anterior representa todo nuestro juego, con todas sus piezas usadas de manera que todo cobra sentido. Pero antes vamos a fijarnos en todo lo que estamos importando en las primeras líneas. Primeramente importamos los módulos `random`, `sys` y `time` que ya vienen definidos por defecto en Python. El módulo `random` nos permite generar números aleatorios (*random* significa aleatorio en inglés). Con el módulo `sys` podremos interactuar con el sistema de nuestro ordenador. Además, como podrás imaginarte, el módulo `time` nos deja manipular el tiempo dentro de nuestro programa.
+
+Luego importamos `pygame` que contiene todas las funcionalidades necesarias para nuestro videojuego. Por último importamos las clases que ya hemos creado en nuestro proyecto: del archivo **fruit.py** dentro del módulo **model** importamos la clase `Fruit`. De manera análoga importamos las clases `Snake`, `Direction`, `Renderer` y `Color`. Ahora veamos cómo engranar todo en la clase `Game`.
+
+## La clase `Game`
+
+Lo primero, como siempre, es el método `__init__`. En este caso no recibiremos ningún parámetro. En la primera línea iniciamos el motor gráfico de Pygame, que es el que nos muestra todo en pantalla. Luego con `pygame.display.set_caption("Snake")` estamos creando el título de nuestro juego, el texto **Snake** se verá en el borde superior de la ventana del juego. A continuación damos valores a la velocidad de juego, el ancho y la altura de la ventana de juego. El campo `game_window` se crea haciendo `self.game_window = pygame.display.set_mode((self.window_x, self.window_y))`. Este representa la ventana donde se desarrolla el juego y es el que tanto se usa en las funciones que ya vimos el día anterior en la clase `Renderer`. Además creamos el campo `fps` (fraps per second) o sea, la cantidad de veces que se dibuja nuestro juego en la pantalla por segundo. Este valor se toma directamente del reloj de Pygame haciendo `self.fps = pygame.time.Clock()`.
```

Comment:
It should be `frames per second`, right?

Reply:
Yes, I don't know where I got fraps from